core/domain/classifier_services_test.py

```diff
@@ -83,3 +87,78 @@ def test_string_classifier_classification(self):
                 self._is_string_classifier_called('abc acb bac bca cbb cba'))
             self.assertTrue(
                 self._is_string_classifier_called('dunno, just guessed'))
+
+    def test_retrieval_of_classifiers(self):
+        """Test the get_classifier_by_id method."""
+
+        with self.assertRaisesRegexp(Exception, (
+            "Entity for class ClassifierModel with id fake_id not found")):
+            classifier_services.get_classifier_by_id('fake_id')
+
+        exp_id = u'1'
+        state = 'Home'
+        classifier_id = classifier_models.ClassifierModel.create(
+            exp_id, 1, state,
+            feconf.INTERACTION_CLASSIFIER_MAPPING['TextInput'], [], 1)
+        classifier = classifier_services.get_classifier_by_id(
+            classifier_id)
+        self.assertEqual(classifier.exp_id, exp_id)
+        self.assertEqual(classifier.state_name, state)
+
+    def test_deletion_of_classifiers(self):
+        """Test the delete_classifier method."""
+
+        with self.assertRaisesRegexp(Exception, (
+            "Entity for class ClassifierModel with id fake_id not found")):
+            classifier_services.delete_classifier('fake_id')
+
+        exp_id = u'1'
+        state = 'Home'
+        classifier_id = classifier_models.ClassifierModel.create(
+            exp_id, 1, state,
+            feconf.INTERACTION_CLASSIFIER_MAPPING['TextInput'], [], 1)
+        classifier_services.delete_classifier(classifier_id)
+        with self.assertRaisesRegexp(Exception, (
+            "Entity for class ClassifierModel with id %s not found" %(
+                classifier_id))):
+            classifier_services.get_classifier_by_id(classifier_id)
+
+    def test_update_of_classifiers(self):
+        """Test the save_classifier method."""
+
+        exp_id = u'1'
+        state = 'Home'
+        test_state = 'State'
```

Comment:
Call this state_name and test_state_name to distinguish from the actual state objects. Ditto above.